File: LV1/1.9/communication/communication.rb
def calc_xor
  d = {}

  256.times do |x|
    a = (x ^ (x << 1)) & 255
    d[a] = x
  end
  d
end

d = calc_xor

_ = gets

l = gets.split.map(&:to_i)
l.map { |i| print "#{d[i]} " }

File: LV1/1.9/communication/communication.py
def calc_xor():
    d = {}

    for x in range(256):
        a = (x ^ (x << 1)) & 255
        d[a] = x
    return d


d = calc_xor()

_ = input()

l = [int(i) for i in input().split(' ')]
[print(d[i], end=" ") for i in l]
